### PR TITLE
Safer fluid evaluation

### PR DESCRIPTION
**Pull request by hnil**
-- safer loop
-- removed asserts which may be triggered in by strange input.

### CODE REVIEW

**Review by atgeirr — APPROVED** [review]
I think this should be merged, but should we add exception throws to replace the asserts?

File: opm/simulators/flow/BlackoilModelEbos.hpp [hunk 1]
```diff
@@ -558,8 +558,6 @@ namespace Opm {
                         Scalar tmpSat = saturationsNew[phaseIdx] - saturationsOld[phaseIdx];
                         resultDelta += tmpSat*tmpSat;
                         resultDenom += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
-                        assert(std::isfinite(resultDelta));
-                        assert(std::isfinite(resultDenom));
```
Comment (bska) [inline opm/simulators/flow/BlackoilModelEbos.hpp:562]: I these are not finite then we have a problem somewhere and I'd rather be told about it.  The mechanism probably should not be `assert()` though.  On another note, PR #4856 changes this area too so we might want to coordinate this with that PR.

**Review by bska — COMMENTED** [review]
Other than the fact that diagnostic messages referring to `InvalideAndUpdateIntensiveQuantitiesOverlap` or similar will be understood only by those who have direct access to the source code I think this looks good.  I agree with @atgeirr though, the check on `isfinite(result*)` should probably remain in place as an exception because if those are not finite then we have a serious problem somewhere.

File: opm/simulators/flow/BlackoilModelEbos.hpp (hunk 1 above)
Comment (atgeirr) [reply opm/simulators/flow/BlackoilModelEbos.hpp:562]: For now, let's leave these two in, and we'll consider it with #4856 instead.

**Comment by bska** [comment]
jenkins build this please

**Pull request merged**